docs/src/examples.md

```diff
@@ -0,0 +1,609 @@
+# column of data, then parse each row as a separate column
+file = CSV.File(IOBuffer(data); transpose=true)
+```
+
+### [Ignoring commented rows](@id comment_example)
+
+```julia
+using CSV
+
+# here, we have several non-data rows that all beging with the "#" string
+data = """
+# row describing column names
+a,b,c
+# row describing first row of data
+1,2,3
+# row describign second row of data
+4,5,6
+"""
+
+# we want to ignore these "commented" rows
+file = CSV.File(IOBuffer(data); comment="#")
+```
+
+### [Ignoring empty rows](@id ignoreemptyrows_example)
+
+```julia
+using CSV
+
+# here, we have a "gap" row in between the first and second row of data
+# by default, these "empty" rows are ignored, but in our case, this is
+# how a row of data is input when all columns have missing/null values
+# so we don't want those rows to be ignored so we can know how many
+# missing cases there are in our data
+data = """
+a,b,c
+1,2,3
+
+4,5,6
+"""
+
+# by passing `ignoreemptyrows=false`, we ensure parsing treats an empty row
+# as each column having a `missing` value set for that row
+file = CSV.File(IOBuffer(data); ignoreemptyrows=true)
+```
+
+### [Including/excluding columns](@id select_example)
+
+```julia
+using CSV
+
+# simple dataset, but we know column "b" isn't needed
+# so we'd like to save time by having parsing ignore it completely
+data = """
+a,b,c
+1,2,3
+4,5,6
+7,8,9
+"""
+
+# there are quite a few ways to provide the select/drop arguments
+# so we provide an example of each, first for selecting the columns
+# "a" and "c" that we want to include or keep from parsing
+file = CSV.File(file; select=[1, 3])
+file = CSV.File(file; select=[:a, :c])
+file = CSV.File(file; select=["a", "c"])
+file = CSV.File(file; select=[true, false, true])
+file = CSV.File(file; select=(i, nm) -> i in (1, 3))
+# now examples of dropping, when we'd rather specify the column(s)
+# we'd like to drop/exclude from parsing
+file = CSV.File(file; drop=[2])
+file = CSV.File(file; drop=[:b])
+file = CSV.File(file; drop=["b"])
+file = CSV.File(file; drop=[false, true, false])
+file = CSV.File(file; drop=(i, nm) -> i == 2)
+```
+
+### [Limiting number of rows from data](@id limit_example)
+
+```julia
+using CSV
+
+# here, we have quite a few rows of data (relative to other examples, lol)
+# but we know we only need the first 3 for the analysis we need to do
+# so instead of spending the time parsing the entire file, we'd like
+# to just read the first 3 rows and ignore the rest
+data = """
+a,b,c
+1,2,3
+4,5,6
+7,8,9
+10,11,12
+13,14,15
+"""
+
+# parsing will start reading rows, and once 3 have been read, it will
+# terminate early, avoiding the parsing of the rest of the data entirely
+file = CSV.File(IOBuffer(data); limit=3)
+```
+
+### [Specifying custom missing strings](@id missing_string_example)
+
+```julia
+using CSV
+
+# in this data, our first column has "missing" values coded with -999
```

Comment:
maybe discuss what is treated as missing by default.
also a particular case is that NOTHING should be treated as missing and the question is how to do it.

Reply:
This is discussed in the `## [`missingstring`](@id missingstring)` section of index.md